test/ComposerOracle.t.sol

```diff
@@ -0,0 +1,191 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.20;
+
+// types
+import {IOracle} from "../src/interfaces/IOracle.sol";
+import {IERC20} from "forge-std/interfaces/IERC20.sol";
+// libraries
+import {console2} from "forge-std/Test.sol";
+// contracts
+import {Test} from "forge-std/Test.sol";
+import {ComposerOracle} from "../src/composer/ComposerOracle.sol";
+import {MockOracle} from "./MockOracle.sol";
+
+contract ComposerOracleTest is Test {
+    // composer oracle
+    ComposerOracle composerOracle;
+
+    // mock chainlink oracle
+    MockOracle chainlinkOracle;
+
+    // token constants
+    address constant DAI = 0x6B175474E89094C44Da98b954EedeAC495271d0F;
+    address constant USDC = 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48;
+    address constant WETH = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
+
+    // raw chainlink prices at about block 19639105
```

Comment:
would we benefit from having all tests run against the same block?

Reply:
That would stop us from adding oracles for data sources that haven't been released yet (e.g. Uniswap v4).